# Virus_Graphs

<img src="Screen Shot 2019-11-04 at 5.56.51 PM.png"
     alt="Markdown Monster icon"
     style="float: left; margin-right: 10px;" />

HIV inference based on NGS data. 

## Ideas

* k-mer idea

HIV inference

n = 170 references
n = 39 without non-recombinants
env, V1-V5
pol

How much variation can we have in a reference graph genome?

Tools?

[VG](https://github.com/vgteam/vg)

[NovoGraph](https://github.com/NCBI-Hackathons/NovoGraph)

[SWIGG](https://github.com/NCBI-Codeathons/SWIGG)

## Outline, plan of action

hey

Day 2: 

Genome graphs with:
T-virus
HIV-1

Datasets: 

SRA, searchterms: "HIV-1 and RNAseq and virus"

Bioproject: PRJNA320293

Unpublished HXB2 ONT dataset.

<img src="igv_snapshot HXB2 4000 read subset.png"
     alt="Markdown Monster icon"
     style="float: left; margin-right: 10px;" />
     
Day 3:

HIV-1 datasets (as mutli-sequence alignments (MSA)/with gaps, and as multi-FASTA/gappless) from [LANL HIV Database](https://www.hiv.lanl.gov/content/sequence/HIV/mainpage.html) November 5th-6th,2019. Files include HXB2 K03455 as reference.

All complete genomes (n=10568)
CD4+ T cell (n=) vs PBMC (n=1555)
Only CCR5 (n= 456) vs only CXCR4 (n= 43) vs R5X4 (n=51)
Acute infection (n=)
(n=)
(n=)
(n=)
(n=)
(n=)
(n=)
(n=)


### Discussions today, 4th November 2019

Too busy..

### Discussions today, 5th November 2019



